File: ruby/hello.rb
def nth_prime(n)
  i=3;
  j=1;
  while :true
    j=j+2;
    if j>i/j
      n=n-1;
      if n==1
        break;
      end
      i=i+2;
      j=1;
    elsif i%j==0
      i=i+2;
      j=1;
    end
  end
  return i;
end
t1 = Time.now
res = nth_prime(300000)
t2 = Time.now
time = ((t2 - t1) * 1000)
puts "答案是#{res},耗时： #{time}ms"

File: cpp/Hello.cpp
#include <ctime>
#include <iostream>
#include <sys/timeb.h>

using namespace std;


long long GetTimeCuoFromTimeb(timeb t)
{
	return t.time * 1000 + t.millitm;
}

unsigned int NthPrime(unsigned int n) {
	unsigned int i = 3, j = 1;
	while (1) {
		j = j + 2;
		if (j * j > i) {
			n = n - 1;
			if (n == 1) {
				break;
			}
			i = i + 2;
			j = 1;
		}
		else if (i % j == 0) {
			i = i + 2;
			j = 1;
		}
	}
	return i;
}

int main() {

	timeb t;
	ftime(&t);//获取毫秒

	unsigned int n = 300000;
	unsigned int result = NthPrime(n);

	timeb t1;
	ftime(&t1);//获取毫秒

	long long time = GetTimeCuoFromTimeb(t1) - GetTimeCuoFromTimeb(t);
	cout << "答案是" << result << "，耗时" << time;
	cin.get();
}

